brokerapi/brokers/account_managers/service_account_manager.go

```diff
@@ -133,11 +134,12 @@ func (sam *ServiceAccountManager) DeleteCredentials(ctx context.Context, binding
 		return fmt.Errorf("Error creating IAM service: %s", err)
 	}
 	saService := iam.NewProjectsServiceAccountsService(iamService)
+	resourceName := projectResourcePrefix + sam.ProjectId + "/serviceAccounts/" + saCreds.UniqueId
+	if _, err = saService.Delete(resourceName).Do(); err != nil {
```

Comment:
Consider scope restricting the err variable to the if block using: if _, err := saService.Delete(resourceName).Do(); err != nil

This approach is better from a defensive programming stand-point. Also consider the same approach for the if blocks throughout this function.